MaxText/configs/models/gemma3-12b.yml

```diff
@@ -21,7 +21,7 @@ base_num_kv_heads: 8
 base_mlp_dim: 15360
 head_dim: 256
 mlp_activations: ["gelu","linear"]
-vocab_size: 262_144
+vocab_size: 262_208
```

Comment:
why do we need this change here an in other model configs?

doesn't this change impact checkpoint checkpoint conversion?

There embedding lookup and unembed layers depend on the vocab size.

Reply:
I reverted the vocab size change for now because changing vocab size requires updating other test scripts, we can do it later with a separate PR if needed. 
But it looks like changing vocab size does not impact the modality of the model because the diff tokens are placeholder image tokens, their embedding will be replaced by real image embedding. And we don't expect to output those tokens. 
So we may not need to change vocab size. I added code to truncate the output logits to a matching size in `forward_pass_logit_checker.py`, with loggings about the truncation. So we can still compare HF vs Maxtext logits.